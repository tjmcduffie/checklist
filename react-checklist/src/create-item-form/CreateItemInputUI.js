import classnames from 'classnames';
import React from 'react';
import './CreateItemInput.css';

const CreateItemInputUI = ({className, onChange, value}) => {
  return (
    <input
      className={classnames('create-item-input', className)}
      id="description"
      onChange={onChange}
      type="text"
      value={value}
    />
  );
}

export default CreateItemInputUI;
