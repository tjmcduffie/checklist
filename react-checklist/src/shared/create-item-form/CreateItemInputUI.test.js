import CreateItemInputUI from './CreateItemInputUI';
import React from 'react';
import ReactDOM from 'react-dom';
import {create} from 'react-test-renderer';

describe('CreateItemInputUI', () => {
  const noop = () => {};

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <CreateItemInputUI className="test" onChange={noop} value="test" />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('renders predictably', () => {
    it('when no className is provided', () => {
      const elem = create(
        <CreateItemInputUI onChange={noop} value="test" />
      );
      const tree = elem.toTree();
      expect(tree.rendered.props.className).toBe('create-item-input');
      expect(elem.toJSON()).toMatchSnapshot();
    });

    it('when a className is provided', () => {
      const elem = create(
        <CreateItemInputUI className="test" onChange={noop} value="test" />
      );
      const tree = elem.toTree();
      expect(tree.rendered.props.className).toBe('create-item-input test');
      expect(elem.toJSON()).toMatchSnapshot();
    });
  });
});
