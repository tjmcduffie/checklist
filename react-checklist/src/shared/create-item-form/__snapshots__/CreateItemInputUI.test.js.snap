// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateItemInputUI renders predictably when a className is provided 1`] = `
<input
  className="create-item-input test"
  id="description"
  onChange={[Function]}
  type="text"
  value="test"
/>
`;

exports[`CreateItemInputUI renders predictably when no className is provided 1`] = `
<input
  className="create-item-input"
  id="description"
  onChange={[Function]}
  type="text"
  value="test"
/>
`;
